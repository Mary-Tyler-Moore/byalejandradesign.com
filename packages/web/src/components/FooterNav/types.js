/** @flow */
export type Props = {
  footerNav: Array<{
    link: string,
    label: string,
  }>,
};
