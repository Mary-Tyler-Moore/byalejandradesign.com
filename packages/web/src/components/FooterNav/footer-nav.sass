@import ../../sass/global.sass

.footerNav
  width: 100%
  display: flex
  justify-content: space-between

  &_link
    @extend %navLink

    color: $white
    margin: 0
    padding: 1rem 0
    margin: 0 1.5rem

    @extend %subtitle-copy-1

    +all-phone
      margin: 0
      padding: 0


    &:hover
      @extend %navHighlight

    &:first-child
      margin-left: 0

    &:last-child
      margin-right: 0
