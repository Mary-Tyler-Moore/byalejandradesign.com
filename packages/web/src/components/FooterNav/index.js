/** @flow */
export { default } from './FooterNav'
export * from './types'

