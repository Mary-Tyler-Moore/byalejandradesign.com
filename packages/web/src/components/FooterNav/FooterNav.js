/** @flow */
import * as React from 'react';
import { Link } from 'gatsby';
import URLToTitle from '../Nav/url-to-title';
// types
import type { Props } from './types';
// styles
import './footer-nav.sass';

const FooterNav = (props: Props) => (
  <nav className="footerNav">
    {props.footerNav.map(({ link, label }) => (
      <React.Fragment>
        <Link
          key={link}
          to={link}
          className="footerNav_link"
          activeClassName="footerNav_link-active"
        >
          {label}
        </Link>
      </React.Fragment>
    ))}
  </nav>
);

export default FooterNav;
