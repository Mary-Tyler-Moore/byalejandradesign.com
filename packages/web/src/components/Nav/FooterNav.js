import * as React from 'react';
import { Link } from 'gatsby';
import { Form } from '@njmyers/component-library';
import Button from '../Button';
import URLToTitle from './url-to-title';
// style
import './footer-nav.sass';

class DesktopFooterNav extends React.Component {
  onSubmit = (e: SyntheticEvent<HTMLFormElement>) => {
    e.preventDefault();

    console.log('clicked');
  };

  render() {
    return (
      <footer className="footer">
        <section className="footer_widgets">
          <form
            name="emailSignup"
            className="emailSignup"
            onSubmit={this.onSubmit}
          >
            <Form.Input
              block="emailSignup"
              name="email"
              label="Signup For Our Newsletter"
              required
            />
            <Button type="submit">Sign Up</Button>
          </form>
        </section>
        <nav className="footerNavDesktop">
          {this.props.footerNav.map((nav) => (
            <Link
              key={nav}
              to={nav}
              className="footerNavDesktop_link"
              activeClassName="footerNavDesktop_link-active"
            >
              {URLToTitle(nav)}
            </Link>
          ))}
        </nav>
      </footer>
    );
  }
}

class MobileFooterNav extends React.Component {
  render() {
    return (
      <nav className="footerNavMobile">
        {this.props.footerNav.map((nav) => (
          <Link
            key={nav}
            to={nav}
            className="footerNavMobile_link"
            activeClassName="footerNavMobile_link-active"
            onClick={this.props.onClick}
          >
            {URLToTitle(nav)}
          </Link>
        ))}
      </nav>
    );
  }
}

const FooterNav = (props) =>
  props.sizes.mobile ? (
    <MobileFooterNav {...props} />
  ) : (
    <DesktopFooterNav {...props} />
  );

export default FooterNav;
