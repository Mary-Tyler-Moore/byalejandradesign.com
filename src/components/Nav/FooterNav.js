import * as React from 'react';
import Link from 'gatsby-link';
import URLToTitle from './url-to-title';
//
import './footer-nav.sass';

class DesktopFooterNav extends React.Component {
  render() {
    return (
      <nav className="footerNavDesktop">
        {this.props.footerNav.map((nav) => (
          <Link
            key={nav}
            to={nav}
            className="footerNavDesktop_link"
            activeClassName="footerNavDesktop_link-active"
          >
            {URLToTitle(nav)}
          </Link>
        ))}
      </nav>
    );
  }
}

class MobileFooterNav extends React.Component {
  render() {
    return (
      <nav className="footerNavMobile">
        {this.props.footerNav.map((nav) => (
          <Link
            key={nav}
            to={nav}
            className="footerNavMobile_link"
            activeClassName="footerNavMobile_link-active"
            onClick={this.props.onClick}
          >
            {URLToTitle(nav)}
          </Link>
        ))}
      </nav>
    );
  }
}

const FooterNav = (props) =>
  props.sizes.mobile ? (
    <MobileFooterNav {...props} />
  ) : (
    <DesktopFooterNav {...props} />
  );

export default FooterNav;
