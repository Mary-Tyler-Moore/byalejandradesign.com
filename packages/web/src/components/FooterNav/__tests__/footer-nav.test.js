import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import FooterNav from '../FooterNav';

describe('it works', () => {
  test('it loads without crashing', () => {
    const render = TestRenderer.create(<FooterNav />);
    expect(render.toJSON()).toMatchSnapshot();
  });

  test('it loads routes crashing', () => {
    const footerNav = [
      {
        link: '/path',
        label: 'Path',
      },
    ];

    const render = TestRenderer.create(<FooterNav footerNav={footerNav} />);
    expect(render.toJSON()).toMatchSnapshot();
  });
});
