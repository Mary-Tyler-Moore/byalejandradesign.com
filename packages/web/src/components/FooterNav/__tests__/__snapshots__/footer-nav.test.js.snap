// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it loads routes crashing 1`] = `
<nav
  className="footerNav"
>
  <a
    className="footerNav_link"
    href="/path"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Path
  </a>
</nav>
`;

exports[`it works it loads without crashing 1`] = `null`;
